docs/language_guide.md

```diff
@@ -792,11 +839,75 @@ Bank :: account <- (accounts) {
 ```
 `withdraw` can be called by any caller which has an account in the bank. The caller's address is bound to the variable `account`, which can then be used in the body of the functions.
 
+### Extended Example
+
+```swift
+contract ChildCollegeFund {
+  var parent: Address
+  var child: Address
+  var canWithdraw: Bool = false
+  var target: Int
+  var contents: Wei = Wei(0)
+}
+// Anyone can initialise the contract.
+// The contract can be deployed by anyone,
+// and is initialised during contract deployment.
+// This initialisation allows you to set the contract address.
+ChildCollegeFund :: (any) {
+  public init(parent: Address, child: Address, target: Int) {
+    self.parent = parent
+    self.child = child
+    self.target = target
+  }
+}
+// Functions both the parent and child can call.
+ChildCollegeFund :: (parent, child) {
+  public func getTarget() -> Int {
+    return target
+  }
+
+  // Trying to withdraw() here would fail as it
+  // requires the caller to be 'child' statically.
+  // At runtime we can only know the caller has one
+  // of the capabilities.
+}
+
+// Functions only the parent can call.
+ChildCollegeFund :: (parent) {
+
+  // The parent can deposit money
+  @payable
+  public mutating func deposit(implicit value: Wei) {
+    contents.transfer(&value)
+  }
 
+  public mutating func allowWithdrawaL() {
+    self.canWithdraw = true
+  }
+
+  public func getContents() -> Int {
+    return contents.getRawValue()
+  }
+
+  public func getDistanceFromGoal() -> Int {
+    // The caller of this function is known to be 'parent'
+    // Therefore, the calls to getTarget and getContents can be performed
+    return getTarget() - getContents() // OK
+  }
+}
+
+// Functions only the child can call
+ChildCollegeFund :: (child) {
+  public mutating func withdraw() {
+    assert(canWithdraw == false)
+    send(child, &contents)
+  }
+}
+```
 ---
 
 ## Payable
-On Ethereum, function calls to other contracts (i.e., "transactions") can be attached with an amount of Wei (the smallest denomination of Ether). Such functions are called "payable". The target account is then credited with the attached amount.
+When a user creates a transaction to call a function, they can attach Ether to send to the contract. Functions which expect Ether to be attached when called must be annotated with the `@payable` annotation. When adding the annotation, a parameter marked implicit of type `Wei` must be declared. `implicit` parameters are a mechanism to expose information from the Ethereum transaction to the developer of the smart contract, without using globally accessible variables defined by the language, such as `msg.value` in Solidity. This mechanism allows developers to name `implicit` variables themselves, and do not need to remember the name of a global variable.
```

Comment:
We should also mention that functions not marked with payable will revert when value is sent to them.